### PR TITLE
[MRG] Test issues

### PR DESCRIPTION
**Pull request by acvander**
- Added pass tests for several functions
- Fail test for train.Saver.restore() causes pytest to fail unexpectedly.

### CODE REVIEW

File: tensorflow-stubs/__init__.pyi [hunk 1]
```diff
@@ -35,97 +81,159 @@ class dtypes:
...
     # single graph element if fetches is a single graph element  OR
     # list of graph elements if fetches is a list of single graph elements OR
     # a dictionary
     # Leaving it as Any for now
-    def run(self, fetches: Any, feed_dict: Optional[Dict[Any, Any]] = None, run_options: Any = None, run_metadata: Any = None) -> Any: ...
+    def run(self, fetches: Any,
+            feed_dict: Optional[Dict[Any, Any]]=...,
+            run_options: Any=...,
+            run_metadata: Any=...
+            ) -> Any: ...
```
Comment (shuttle1987) [inline tensorflow-stubs/__init__.pyi]: I don't think I like throwing away the information about what the default values are. I see that the defaults were replaced with ellipsis (`...`) in these

File: tensorflow-stubs/__init__.pyi [hunk 2]
```diff
@@ -35,97 +81,159 @@ class dtypes:
...
+          out_type: Any=...
+          ) -> Any: ...
 
 # Original function definition for truncated_normal here:
 # https://github.com/tensorflow/tensorflow/blob/70cd9ed2d2ea37a6da6f813a99b32c03e90736a4/tensorflow/python/ops/random_ops.py#L139
-def truncated_normal(shape: Any, mean: Any=0.0, stddev: Any=1.0, dtype: Any=dtypes.float32, seed: Any=None, name: Optional[str]=None) -> Any: ...
+
+
+def truncated_normal(shape: Any,
+                     mean: Any=...,  # default 0.0
```
Comment (shuttle1987) [inline tensorflow-stubs/__init__.pyi]: What's the reasoning behind the change here to `Any=...` with a comment of what the default was as opposed to keeping the original `Any=0.0` as per the default in the source?

File: tests/pass/nn.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import tensorflow as tf
+
+# setup for ctc_beam_search_decoder
+placeholder = tf.placeholder(tf.float32, shape=(100, 256, 10))
+# TODO: pytest gives error "module has no attribute "SparseTensor"
```
[the remark this answers is not in the document]
Comment (shuttle1987) [reply tests/pass/nn.py]: Just to be clear, this test should be merged in, anything that's a valid TF code that fails a test indicates we have to improve the stubs.

[the remark this answers is not in the document]
Comment (acvander) [reply tests/pass/nn.py]: Sounds good. I have uncommented the test in [a77ca05](https://github.com/persephone-tools/tensorflow-stubs/pull/6/commits/a77ca05839e84fb777bb11f3cc2ee4fca26a0df4).

[the remark this answers is not in the document]
Comment (shuttle1987) [reply tests/pass/nn.py]: We can fix #1 after this is merged

File: setup.py [hunk 4]
```diff
@@ -20,5 +31,6 @@ def find_stubs(package):
     packages=["tensorflow-stubs"],
     # PEP 561 requires these
     install_requires=["tensorflow"],
+    extras_requires=extras,
```
[the remark this answers is not in the document]
Comment (acvander) [reply setup.py]: I see how this is meant to be used now. I was trying to minimize the number of setup files being used. 

How do you want to go about dealing with these? 
- Would it be best to utilize Pipfile on top of setup.py?
    - Pipfile does allow for easy venv generation using [Pipenv](https://docs.pipenv.org/).
    - For pipfile, do I need to commit both Pipfile and Pipfile.lock or just Pipfile.lock?
    - Can the tests-requirements.txt then be merged into Pipfile?

[the remark this answers is not in the document]
Comment (shuttle1987) [reply setup.py]: Yeah let's just use a pipfile for everything, it's a better approach to package management for sure (I mean this in a Pareto-efficiency sense, it's just better). I just used a really basic requirements file just to get started with this but I think your way is better. So let's just remove test-requirements.txt and put it in the `[dev-packages]` section of the pipfile

File: tensorflow-stubs/__init__.pyi (hunk 2 above)
Comment (acvander) [reply tensorflow-stubs/__init__.pyi]: I was following [PEP 484](https://www.python.org/dev/peps/pep-0484/#id37).

File: setup.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (acvander) [reply setup.py]: Pipfile has been added for everything in [5675501](https://github.com/persephone-tools/tensorflow-stubs/pull/6/commits/567550141196fb7ffc219bfb802fa42fca52077a). I also had to modify the travis.yml to account for these changes.

**Comment by acvander** [comment]
I have updated everything with the new pipfile and travis-ci setup, but the tests are now throwing the following errors
`
E       assert "tensorflow-s...ribute 'nn'\n" == ''

E         - tensorflow-stubs/__init__.pyi:3: error: Module 'tensorflow' has no attribute 'errors'\
E         - tensorflow-stubs/__init__.pyi:4: error: Module 'tensorflow' has no attribute 'train'\
E         - tensorflow-stubs/__init__.pyi:5: error: Module 'tensorflow' has no attribute 'nn'\
`

File: tensorflow-stubs/__init__.pyi (hunk 2 above)
Comment (shuttle1987) [reply tensorflow-stubs/__init__.pyi]: Looking at this more, this seems reasonable.

**Comment by shuttle1987** [comment]
This may be an issue with Travis CI interacting with pipenv, I'll have to look at this tomorrow when I can pull the code and investigate further.

**Comment by acvander** [comment]
I've got the above issue with travis sorted out. It was an issue of not installing the local package directory via the Pipfile. 

In the process of fixing it, I added a Dockerfile to create a clean container to run the tests. It gave the same response as travis prior to fixing the issue.

Finally, I got rid of the Makefile and combined it all into travis.yml.

File: tensorflow-stubs/__init__.pyi (hunk 2 above)
Comment (shuttle1987) [reply tensorflow-stubs/__init__.pyi]: Interestingly enough the PEP got changed recently, so these changes are a good way to deal with it.

**Pull request merged**